src/Symfony/Component/Form/FormFactory.php

```diff
@@ -102,10 +103,36 @@ public function createBuilderForProperty($class, $property, $data = null, array
             return $this->createNamedBuilder($property, 'text', $data, $options);
         }
 
-        $typeGuess = $guesser->guessType($class, $property);
-        $maxLengthGuess = $guesser->guessMaxLength($class, $property);
-        $requiredGuess = $guesser->guessRequired($class, $property);
-        $patternGuess = $guesser->guessPattern($class, $property);
+        $typeGuess = null;
+        $maxLengthGuess = null;
+        $requiredGuess = null;
+        $patternGuess = null;
+
+        if (!empty($options['property_path'])) {
+            $propertyPath = new PropertyPath($options['property_path']);
+            $subProperty = null;
+
+            foreach ($propertyPath as $i => $subProperty) {
+                if ($propertyPath->isIndex($i)) {
+                    continue;
+                }
+
+                $typeGuess = $guesser->guessType($class, $subProperty);
+
+                if ($typeGuess->getType() === 'Symfony\Bridge\Doctrine\Form\Type\EntityType') {
+                    $class = $typeGuess->getOptions()['class'];
+                }
```

Comment:
Is it enough to check for Symfony\Bridge\Doctrine\Form\Type\EntityType here? Currently, if one sub property is not an EntityType, then the guess could be wrong.
Any idea?